applications/tari_dan_wallet_daemon/src/services/template_monitor.rs

```diff
@@ -42,6 +46,35 @@ where
         }
     }
 
+    /// Fetching template definition with retry.
+    async fn fetch_template_definition(&self, template_address: TemplateAddress) -> anyhow::Result<TemplateDef> {
+        let min_wait_time = Duration::from_millis(100);
+        let max_wait_time = Duration::from_secs(5);
+        let wait_step = Duration::from_millis(100);
+        let mut current_wait_time = min_wait_time;
+        let mut template_definition = None;
+        let network_interface = self.wallet_sdk.get_network_interface();
+        while template_definition.is_none() {
+            template_definition = match network_interface
+                .fetch_template_definition(template_address)
+                .await
+                .map_err(|error| TransactionApiError::NetworkInterfaceError(format!("{}", error)))
+            {
+                Ok(template_def) => Some(template_def),
+                Err(error) => {
+                    error!(target: LOG_TARGET, "Failed to fetch template definition: {}, retry...", error);
+                    tokio::time::sleep(current_wait_time).await;
```

Comment:
May want to add a check for shutdown so that this loop will exit <= 1s if shutdown
```suggestion
if self.shutdown_signal.is_triggered() {
return Err(anyhow!("shutdown during fetch template definition"));
}
```

Reply:
ah true makes sense

Reply:
pushed